lib/app.rb

```diff
@@ -0,0 +1,28 @@
+require 'sinatra/base'
+# require 'byebug'
+# require 'pry'
+require 'tilt/erb'
+require 'data_mapper'
+require 'dm-migrations'
+
+class App < Sinatra::Base
+  # helpers Sinatra::FormHelpers
+  set :views, proc {File.join(root, '..', 'views')}
+  enable :sessions
+  set :session_secret, '121212121212'
+  use Rack::Session::Pool
+  env = ENV['RACK_ENV'] || "development"
+  DataMapper.setup(:default, ENV['DATABASE_URL'] || "postgres://localhost/ohchit_#{env}")
+
+  # DataMapper::Logger.new($stdout, :debug)
+  DataMapper::Model.raise_on_save_failure = true
+  DataMapper.finalize
+  #DataMapper.auto_migrate!
+  DataMapper.auto_upgrade!
+  #byebug
+  get '/' do
+    erb :index
+  end
+  # start the server if ruby file executed directly
+  run! if app_file == $0
+end
```

Comment:
Again, did you copy and paste this code Chris? The feeling I get is that a lot of this settings are copied from one of my demo apps or the weeks challenge. It is okay to read other peoples code. To copy and paste is not. We learn by doing.